addons/binding/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/handler/KonnectedHandler.java

```diff
@@ -0,0 +1,476 @@
+
+    /**
+     * This is the constructor of the Konnected Handler.
+     *
+     * @param thing          the instance of the Konnected thing
+     * @param webHookServlet the instance of the callback servlet that is running for communication with the Konnected
+     *                           Module
+     * @param hostAddress    the webaddress of the OpenHAB server instance obtained by the runtime
+     * @param port           the port on which the OpenHAB instance is running that was obtained by the runtime.
+     */
+    public KonnectedHandler(Thing thing, KonnectedHTTPServlet webHookServlet, String hostAddress, String port) {
+        super(thing);
+        this.webHookServlet = webHookServlet;
+        callbackIpAddress = hostAddress + ":" + port;
+        logger.debug("The callback ip address is: {}", callbackIpAddress);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // get the zone number in integer form
+        Integer zone = Integer.parseInt(channelUID.getId().substring((channelUID.getId().length() - 1)));
+        logger.debug("The channelUID is: {} and the zone is :", channelUID.getAsString(), zone);
+        // convert the zone to the pin based on value at index of zone
+        Integer pin = Arrays.asList(PIN_TO_ZONE).get(zone);
+        logger.debug("Zachary test {}", command instanceof OnOffType);
+        // if the command is OnOfftype
+        if (command instanceof OnOffType) {
+            int sendCommand = (OnOffType.OFF.compareTo((OnOffType) command));
+            logger.debug("The command being sent to pin {} for channel:  is {}", channelUID.getAsString(), pin,
+                    sendCommand);
+            sendActuatorCommand(Integer.toString(sendCommand), pin, channelUID);
+        } else if (command instanceof OpenClosedType) {
+            logger.debug("A command was sent to a sensor type so we are ignoring the command");
+        }
+    }
+
+    /**
+     * Process a {@link WebHookEvent} that has been received by the Servlet from a Konnected module with respect to a
+     * sensor event
+     *
+     * @param pin     the pin number which which was activated
+     * @param state   the state of the pin
+     * @param setAuth the token sent with the response
+     */
+    public void handleWebHookEvent(KonnectedModuleGson event, String sentAuth) {
+        // if we receive a command upteate the thing status to being online
+        updateStatus(ThingStatus.ONLINE);
+        // get the zone number based off of the index location of the pin value
+        String sentZone = Integer.toString(Arrays.asList(PIN_TO_ZONE).indexOf(event.getPin()));
+        // check that the zone number is in one of the channelUID definitions
+        getThing().getChannels().forEach(channel -> {
+            ChannelUID channelId = channel.getUID();
+            String zoneNumber = channelId.getId().substring((channelId.getId().length() - 1));
```

Comment:
Oh oh. The channelId can be ANYTHING - you must not derive any semantics from it. Might you be lacking a "zone" parameter on the channel types?

Reply:
The only semantic that matters is the number at the end which corresponds to the physical location on the konnected board see https://github.com/openhab/openhab2-addons/pull/3702#discussion_r232340490

other than that from a development prospective i don't care what the channels are called and in fact i cant understand why people might want different names.  For example if they have an actuator hooked up to zone 7. They could have multiple channels that all end in 7 with different timing configs.  Then channels ids could then be names things like "long7" "short7" "stucato7" etc. and the binding will process them all exactly the same and be none the wiser but the user can easily identify which channel means what to him.

Reply:
Please move the zone number to a config parameter.

Reply:
done